### DISCUSSION TITLE
_r hud animations

### DISCUSSION THREAD

**Discussion by romanpopulik** [OGSR/OGSR-Engine#304] [Q&A] [answered]
I couldn't find anything about the _r animations stored in weapons hud ogf's. Are they used anyhow in game ? If not, is it possible to remove them from engine, so they will not cause crash anymore ?

**Comment by Xottab-DUTY** [comment]
If you will look at original SOC's weapons, you'll find it. (or you'll find it in the original .object's)

I think GSC planned to use them in `two weapons in two hands` feature (like, two pistols in both hands), but never were actually used. (maybe in non-leaked builds they were?..)

**Comment by romanpopulik** [comment]
I know they are in original huds. I just think it would be great if lack of them won't crash the game.

**Comment by xrSimpAI** [comment, marked as answer]
> I couldn't find anything about the _r animations stored in weapons hud ogf's. Are they used anyhow in game ?

Yes, they are used when the actor is driving a car:
```cpp
CWeaponPistol::WWPMotions&	 CWeaponPistol::wwpm_current	()
{
	CActor* A = smart_cast<CActor*>(H_Parent());
	if(A && A->Holder()){	
//		Msg("right-hand animation playing");
		return				mhud_pistol_r;
	}
//	Msg("double-hands animation playing");
	return					mhud_pistol;
}
```

**Reply by romanpopulik** [reply]
So they must stay. Moreoveer, probably they should look in original (left hand not used, just stay down). I'm curious how will work in car _r anims which I use in custom weapons models (I simply copy all normal anims and add _r suffix).

**Reply by xrSimpAI** [reply]
They will work in the same way as outside of transport.

**Reply by romanpopulik** [reply]
So they can stay the same as normal anims. You just need to handle a steering wheel with your knees ;-)
